DETAILS.md

```diff
@@ -0,0 +1,115 @@
+
+Athens will ship with a CLI to construct and upload the zip file.
+
+# How Will Modules Be Served?
+
+There are two modes of serving modules. One involves using the `GOPROXY` environment variable,
+and the other doesn't.
+
+## Using `GOPROXY`
+
+When a user sets `GOPROXY` on their system, the `vgo` tool automatically prefixes all modules
+with that value when it makes requests. For example,
+`GOPROXY=localhost:3000 go get my/thing` will execute the download protocol against
+`localhost:3000`.
+
+With this in mind, we'll give users two `GOPROXY` related workflows.
+
+### Workflow I - Local Development
+
+We'll give users a CLI that runs a proxy server that passes through to a tree on their
+local filesystem. The workflow will be the following:
+
+```console
+$ athens run local-proxy --root /someplace --vcs git
+$ GOPROXY=localhost:1234 go get my/module
+```
+
+The `go get` command executes the download protocol on the local proxy, which simply mirrors
+to the files in `/someplace`. In the case of `my/module`, it mirrors to `/someplace/my/module`, requires the `go.mod` to be there, and uses `git` (specified by `--vcs`) to list
+tags, etc...
+
+### Workflow II - Proxying to Github Repositories
+
+We'll give users the ability to `go get` modules against our `gomods.io` server that are
+really mirrored from Github. In the webhook flow (above), we already have the capability
+to fetch code from a Github repository, so we can know the `github.com/...` package name.
+
+So, we can make the following work properly:
+
+```console
+$ GOPROXY=gomods.io go get github.com/arschles/thing@v1
+```
+
+This not only executes the download protocol against `gomods.io`, it also means that the
+import path for the module is `github.com/arschles/thing`, and that will certainly help
+people migrate their codebases from previous dependency managers to vgo.
+
+I'd prefer that our registry serves the download protocol directly (using a `<meta>` tag redirect, more on that below)
+so that we can still serve from our CDN and protect against repositories disappearing. I like the idea technically, but
+I don't know about it from a community perspective.
+
+## "Vanity" Module Names
+
+We've been able to specify "vanity" names (names that don't match the repository the
+code lives in) for our packages for a while not. https://gopkg.in was one of the earliest
+systems that let us do this. It "redirected" (using the `<meta>` tag) the `go get`
+requests against it to the appropriate repository in Github. The `go get` tool would
+then follow the redirect and use `git` to fetch the code.
+
+Now that there's a custom HTTP-based download protocol, we can use the same rename-by-redirecting mechanism, but with the following additions:
+
+* Allow users to specify their own "vanity" module names for a given codebase - We should probably allow 0 or more vanity names attached to an existing repository - We should allow users to turn off the repository-based proxy name (i.e. the `github.com`
```

Comment:
s/0 or more/0 or 1
unless people are masochistic enough to want to publish the same package with different import names.

Reply:
yea, good point. let's limit to 1